### DISCUSSION TITLE
Adonis V5 provider depending on another provider.

### DISCUSSION THREAD

**Discussion by Xstoudi** [adonisjs/core#1570] [Help]
Hello!

I'm currently developing an Adonis5 provider that contains a class that needs to extend a class from another provider.

Currently, the code of MY provider is the following :
```ts
import { IocContract } from '@adonisjs/fold'
import { YYY } from '../YYY'

export default class UserProvider {
  constructor (protected container: IocContract) {
  }

  public register () {
    this.container.singleton('XXX/YYY', () => {
      return new YYY(this.container)
    })
  }

  public boot () {
    // IoC container is ready

  }

  public shutdown () {
    // Cleanup, since app is going down
  }

  public ready () {
    // App is ready
  }
}
```

The `YYY` class is a singleton that instantiate class `ZZZ`. The `ZZZ` class needs to extend class from another provider.

The problem is that right now, importing  the parent class from another provider leads to an `IocLookupException`.

How can I ensure that the parent class is already known by the IoC container before importing it?

**Comment by thetutlage** [comment]
Can you share the code that attempts to extend the class?

**Comment by Xstoudi** [comment]
It is literally `class XXX extends YYY {}` but that not the core problem. 
I'm gonna create a repro repo containing what I would like to do and what I found to monkey-patch.
